hudi-io/src/main/java/org/apache/hudi/io/hfile/HFileCompressionCodec.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.io.hfile;
+
+import org.apache.hudi.io.compress.CompressionCodec;
+
+/**
+ * Compression codec supported by HFile.
+ * <p>
+ * The ordinal of these cannot change or else that breaks all existing HFiles out there,
+ * even the ones that are not compressed! (They use the NONE algorithm)
+ * This is because HFile stores the ordinal to indicate which compression codec is used.
+ */
+public enum HFileCompressionCodec {
```

Comment:
Can you eliminate this class? Just feels like it as indirection

Reply:
The ordinal of the codec from this class is used in the HFile to indicate the compression codec so I feel we still need this instead of coupling with `CompressionCodec` class.

Reply:
This class is deleted and replaced by `HFileUtils#decodeHFileCompressionCodec`.